### PR TITLE
Model benchmark: Added support for Semantic Segmentation. Refactor Object Detection, Instance Segmentation, Model comparison.

### PR DESCRIPTION
**Pull request by almazgimaev**
- Added **support for semantic segmentation**
- **refactor code base** for `Object Detection`, `Instance Segmentation`, `Model comparison`

additional needed changes:
- new argument `extra_validation` in `sly.Bitmap`
- fix `_predictions_to_annotation` and `_create_label` methods  for semantic segmentation task
- added `default_bg_name` and `default_bg_color` arguments in 
- `sly.Project. to_segmentation_task` updated to support local project without saved images
- added progress in `TagApi.add_to_objects`

### CODE REVIEW

File: supervisely/project/project.py [hunk 1]
```diff
@@ -2281,6 +2282,8 @@ def to_segmentation_task(
         target_classes: Optional[List[str]] = None,
         progress_cb: Optional[Union[tqdm, Callable]] = None,
         segmentation_type: Optional[str] = "semantic",
+        default_bg_name: Optional[str] = "__bg__",
```
Comment (almazgimaev) [inline supervisely/project/project.py:2285]: `default_bg_name` > `bg_name`

File: supervisely/nn/benchmark/base_benchmark.py [hunk 2]
```diff
@@ -130,6 +147,12 @@ def _run_inference(
             "batch_size": batch_size,
             **model_info,
         }
+        if self.train_info:
+            inference_info["train_info"] = self.train_info
+        if self.evaluator_app_info:
+            inference_info["evaluator_app_info"] = self.evaluator_app_info
+        if self.gt_images_ids:
+            inference_info["gt_images_ids"] = self.gt_images_ids
```
Comment (almazgimaev) [inline supervisely/nn/benchmark/base_benchmark.py:155]: do not save it in inference_info JSON

File: supervisely/nn/benchmark/base_benchmark.py [hunk 3]
```diff
@@ -450,8 +466,16 @@ def visualize(self, dt_project_id=None):
...
             self.dt_project_info = self.api.project.get_info_by_id(dt_project_id)
 
-        vis = Visualizer(self)
-        vis.visualize()
+        if self.visualizer_cls is None:
+            raise RuntimeError(
+                f"Visualizer class is not defined in {self.__class__.__name__}. "
+                "It should be defined in the subclass of BaseBenchmark (e.g. ObjectDetectionBenchmark)."
+            )
+        eval_result = self.evaluator.get_eval_result()
```
Comment (almazgimaev) [inline supervisely/nn/benchmark/base_benchmark.py:474]: - make `inference_info` optional and move `EvalResut` reading to visualizer
- mb rename `EvalResut`?

File: supervisely/nn/benchmark/utils/semantic_segmentation/table_template.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
+table_template = """
```
Comment (almazgimaev) [inline supervisely/nn/benchmark/utils/semantic_segmentation/table_template.py:1]: mb remove?

File: supervisely/nn/benchmark/utils/utils.py [hunk 5]
```diff
@@ -3,6 +3,7 @@
 WORKSPACE_NAME = "Model Benchmark: predictions and differences"
 WORKSPACE_DESCRIPTION = "Technical workspace for model benchmarking. Contains predictions and differences between ground truth and predictions."
 
+
 def try_set_conf_auto(session: SessionJSON, conf: float):
```
Comment (almazgimaev) [inline supervisely/nn/benchmark/utils/utils.py:7]: move to `coco` dir and rename coco dir > `detection`

File: supervisely/nn/benchmark/comparison/sem_seg_model_comparison.py [hunk 6]
```diff
@@ -0,0 +1,356 @@
+import supervisely as sly
```
Comment (almazgimaev) [inline supervisely/nn/benchmark/comparison/sem_seg_model_comparison.py:1]: is it work?

**Review by GoldenAnpu — APPROVED** [review]

**Pull request merged**